Open-Lineage type-ignore in OpenLineageRedactor likely hides some problem

### Body

The new `attrs` package 23.1 released on 16th of April (11 hours ago) added typing information to "attrs.asdict" method and it mypy tests started to fail with 

```
airflow/providers/openlineage/utils/utils.py:345: error: Argument 1 to "asdict"
has incompatible type "Type[AttrsInstance]"; expected "AttrsInstance" 
[arg-type]
    ...            for dict_key, subval in attrs.asdict(item, recurse=False)....
                                                        ^
airflow/providers/openlineage/utils/utils.py:345: note: ClassVar protocol member AttrsInstance.__attrs_attrs__ can never be matched by a class object
```

The nature of this error (receiving Type where expecting instance indicates that there is somewhat serious issue here. 
Especially that there were a `type: ignore` one line above that would indicate that something is quite wrong here (when we ignore typing issue, we usually comment why and ignore very specific error (`type: ignore[attr-undefined]` for example) when we have good reason to ignore it.

Since open-lineage is not yet released/functional and partially in progress, this is not an issue to be solved immediately, but soon (cc: @mobuchowski).

For now I am workaroudning this by adding another `type: ignore` to stop the static checks from failing (they fail only for PRs that are updating dependencies) and allow to upgrade to attrs 23.1.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

--- a/airflow/providers/openlineage/utils/utils.py
+++ b/airflow/providers/openlineage/utils/utils.py
@@ -341,10 +341,8 @@ class OpenLineageRedactor(SecretsMasker):
         try:
             if name and should_hide_value_for_key(name):
                 return self._redact_all(item, depth, max_depth)
-            # TODO: Those type: ignores here should be reviewed and fixed
-            # See: https://github.com/apache/airflow/issues/30673
-            if attrs.has(item):  # type: ignore
-                for dict_key, subval in attrs.asdict(item, recurse=False).items():  # type: ignore
+            if attrs.has(type(item)):
+                for dict_key, subval in attrs.asdict(item, recurse=False).items():
                     if _is_name_redactable(dict_key, item):
                         setattr(
                             item,

